lib/rubocop/cop/rspec/expect_actual.rb

```diff
@@ -49,6 +49,17 @@ def on_send(node)
           end
         end
 
+        def autocorrect(node)
+          lambda do |corrector|
+            expectation = node.parent.parent
+            rhs = expectation.children.last
+            return unless rhs.is_a?(RuboCop::AST::MethodDispatchNode)
+            return if rhs.method_name != :eq
+
+            swap_order(corrector, node, rhs.children.last)
+          end
+        end
```

Comment:
Played around with it, and came up with this:
```ruby
        def_node_matcher :expect_literal, <<~PATTERN
          (send
            (send nil? :expect $#literal?)
            #{Runners::ALL.node_pattern_union}
            $(send nil? ...)
          )
        PATTERN

        def on_send(node)
          expect_literal(node) do |argument|
            add_offense(node, location: argument.source_range)
          end
        end

        def autocorrect(node)
          argument, matcher = expect_literal(node)
          lambda do |corrector|
            return if matcher.method_name != :eq

            swap_order(corrector, argument, matcher.arguments.first)
          end
        end
```

Node pattern is hard to grasp at first but helps to get rid of type checks and node deconstruction/walking.

Up to you to keep as as or cherry-pick parts of that code.